### PR TITLE
Feature Flag Plugin Alpha version

### PR DESCRIPTION
**Pull request by Debanitrkl**
## Describe your changes
Added the Feature Flag plugin
## Checklist
- [ ] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.

### CODE REVIEW

**Comment by Debanitrkl** [comment]
cc @knagurski for review

File: plugins/harness-ci-cd/src/components/ExecutionList/ExecutionList.tsx [hunk 1]
```diff
@@ -1,3 +1,5 @@
+/* eslint-disable import/no-extraneous-dependencies */
+/* eslint-disable @typescript-eslint/no-shadow */
```
Comment (jenil-harness) [inline plugins/harness-ci-cd/src/components/ExecutionList/ExecutionList.tsx:2]: Rather than disabling let's fix this.

Comment (Debanitrkl) [reply plugins/harness-ci-cd/src/components/ExecutionList/ExecutionList.tsx]: Thanks for identifying, they were already fixed removed it from the code, please check the latest commit.

File: packages/app/src/components/catalog/EntityPage.tsx [hunk 2]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable import/no-extraneous-dependencies */
```
[the remark this answers is not in the document]
Comment (Debanitrkl) [reply packages/app/src/components/catalog/EntityPage.tsx]: Removed this as well in the recent commit.

File: plugins/harness-ci-cd/src/components/ExecutionList/ExecutionList.tsx [hunk 3]
```diff
@@ -956,6 +956,7 @@ function ExecutionList() {
       });
     }
     if (toggle) {
+      // eslint-disable-next-line @typescript-eslint/no-shadow
       const response = await fetch(
```
Comment (jenil-harness) [inline plugins/harness-ci-cd/src/components/ExecutionList/ExecutionList.tsx:960]: you can rename `response` to maybe `res` or `apiRes` to resolve this eslint issue.

File: plugins/harness-feature-flags/README.md [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+
+```tsx
+// In packages/app/src/components/catalog/EntityPage.tsx
+
+import {
+  isHarnessFeatureFlagAvailable,
+  EntityHarnessFeatureFlagContent,
+} from '@internal/plugin-harness-feature-flags';
+
+const featureFlagList = (
```
Comment (jenil-harness) [inline plugins/harness-feature-flags/README.md:63]: Should we also include mounting it on the path?
```
    <EntityLayout.Route path="/feature-flag" title="Feature Flags">
      {featureFlagList}
    </EntityLayout.Route>
```

File: plugins/harness-feature-flags/src/route-refs.tsx [hunk 5]
```diff
@@ -0,0 +1,11 @@
+import { createRouteRef, createSubRouteRef } from '@backstage/core-plugin-api';
+
+export const harnessFeatureFlagsRouteRef = createRouteRef({
```
Comment (jenil-harness) [inline plugins/harness-feature-flags/src/route-refs.tsx:3]: This is also present in `routes.tsx` why do we need to redefine here?

File: plugins/harness-feature-flags/src/components/FeatureList/FeatureList.tsx [hunk 6]
```diff
@@ -0,0 +1,358 @@
...
+  const backendBaseUrl = discoveryApi.getBaseUrl('proxy');
+  const config = useApi(configApiRef);
+  const baseUrl =
+    config.getOptionalString('harness.baseUrl') ?? 'https://app.harness.io/';
+
+  const { projectId, orgId, accountId, urlParams } = useProjectSlugFromEntity();
+
+  async function getFeatureEnv() {
+    const resp = await fetch(
+      `${await backendBaseUrl}/harness/gateway/ng/api/environments?accountId=${accountId}&outingId=${accountId}&orgIdentifier=${orgId}&projectIdentifier=${projectId}`,
```
Comment (jenil-harness) [inline plugins/harness-feature-flags/src/components/FeatureList/FeatureList.tsx:79]:
```suggestion
      `${await backendBaseUrl}/harness/gateway/ng/api/environments?accountId=${accountId}&routingId=${accountId}&orgIdentifier=${orgId}&projectIdentifier=${projectId}`,
```

File: plugins/harness-feature-flags/src/components/FeatureList/FeatureList.tsx [hunk 7]
```diff
@@ -0,0 +1,358 @@
...
+      orgIdentifier: `${orgId}`,
+    });
+    if (done) {
+      if (!envId) {
+        setEnvId(envIds[0]);
+      } else {
+        // eslint-disable-next-line no-inner-declarations
+        async function fetchFeatures() {
+          const resp2 = await fetch(
+            `${await backendBaseUrl}/harness/gateway/cf/admin/features?${query}&metrics=true&flagCounts=true&name=&summary=true`,
```
Comment (jenil-harness) [inline plugins/harness-feature-flags/src/components/FeatureList/FeatureList.tsx:218]: `name` queryParam is empty on purpose?

File: plugins/harness-feature-flags/src/components/FeatureList/FeatureList.tsx [hunk 8]
```diff
@@ -0,0 +1,358 @@
...
+            else if (resp2.status === 401) setState(AsyncStatus.Unauthorized);
+            else setState(AsyncStatus.Error);
+          }
+          const data = await resp2.json();
+          if (data.itemCount < 50) {
+            setTotalElements(data.itemCount);
+          }
+          const getFeatureList = (): Array<{}> => {
+            const data1: Array<TableData> = [];
+            while (data1.length < data.itemCount)
```
Comment (jenil-harness) [inline plugins/harness-feature-flags/src/components/FeatureList/FeatureList.tsx:231]: let's use `map` here.
```
const data1 = data.features.slice(0,50).map( d => ({
...
}))
```

File: plugins/harness-ci-cd/src/components/ExecutionList/ExecutionList.tsx (hunk 3 above)
Comment (Debanitrkl) [reply plugins/harness-ci-cd/src/components/ExecutionList/ExecutionList.tsx]: Done, removed this disable

File: plugins/harness-feature-flags/src/components/FeatureList/FeatureList.tsx (hunk 6 above)
Comment (Debanitrkl) [reply plugins/harness-feature-flags/src/components/FeatureList/FeatureList.tsx]: done

File: plugins/harness-feature-flags/src/components/FeatureList/FeatureList.tsx (hunk 7 above)
Comment (Debanitrkl) [reply plugins/harness-feature-flags/src/components/FeatureList/FeatureList.tsx:218]: declared a const and passed the function, removed the no-inner-declarations

File: plugins/harness-feature-flags/src/components/FeatureList/FeatureList.tsx (hunk 8 above)
Comment (Debanitrkl) [reply plugins/harness-feature-flags/src/components/FeatureList/FeatureList.tsx]: Added a map, replaced all "any"

File: plugins/harness-feature-flags/src/components/FeatureList/FeatureList.tsx [hunk 9]
```diff
@@ -0,0 +1,363 @@
...
+      }
+    }
+  }, [refresh, envId]);
+
+  const handleChange = (event: SelectChangeEvent) => {
+    setEnvId(event.target.value as string);
+    setState(AsyncStatus.Success);
+  };
+
+  // if (
```
Comment (jenil-harness) [inline plugins/harness-feature-flags/src/components/FeatureList/FeatureList.tsx:268]: Remove uncommented code.

Comment (Debanitrkl) [reply plugins/harness-feature-flags/src/components/FeatureList/FeatureList.tsx]: Done

**Review by jenil-harness — APPROVED** [review]

**Review by sandeepa-kv-jois — APPROVED** [review]

**Pull request merged**